awx/main/utils/handlers.py

```diff
@@ -24,24 +25,22 @@ def _connect_unixsocket(self, address):
     def emit(self, msg):
         if not settings.LOG_AGGREGATOR_ENABLED:
             return
-        if not os.path.exists(settings.LOGGING['handlers']['external_logger']['address']):
-            return
         try:
             return super(RSysLogHandler, self).emit(msg)
-        except ConnectionRefusedError:
-            # rsyslogd has gone to lunch; this generally means that it's just
-            # been restarted (due to a configuration change)
-            # unfortunately, we can't log that because...rsyslogd is down (and
-            # would just us back ddown this code path)
-            pass
-        except BlockingIOError:
-            # for <some reason>, rsyslogd is no longer reading from the domain socket, and
-            # we're unable to write any more to it without blocking (we've seen this behavior
-            # from time to time when logging is totally misconfigured;
-            # in this scenario, it also makes more sense to just drop the messages,
-            # because the alternative is blocking the socket.send() in the
-            # Python process, which we definitely don't want to do)
-            pass
+        except (FileNotFoundError, ConnectionRefusedError, BlockingIOError) as e:
+            # for any number of reasons, rsyslogd has gone to lunch;
+            # this usually means that it's just been restarted (due to
+            # a configuration change) unfortunately, we can't log that
+            # because...rsyslogd is down (and would just put us back down this
+            # code path)
+            # as a fallback, it makes the most sense to just write the
+            # messages to sys.stderr (which will end up in supervisord logs,
+            # and in containerized installs, cascaded down to pod logs)
+            # because the alternative is blocking the
+            # socket.send() in the Python process, which we definitely don't
+            # want to do)
+            sys.stderr.write(e)
```

Comment:
I'm working on testing this, and trying to come up with a reproducible way to actually _hit_ `ConnectionRefusedError` or `BlockingIOError`, but so far I haven't been able to throw enough load at rsyslogd to get it to behave this way.

Reply:
From https://rsyslog.readthedocs.io/en/latest/configuration/modules/imuxsock.html:

```Unix log sockets can be flow-controlled. That is, if processing queues fill up, the unix socket reader is blocked for a short while. This may be useful to prevent overruning the queues (which may cause excessive disk-io where it actually would not be needed).```

So this might be one way to accomplish it, but this behavior is *off* by default in rsyslogd.

Reply:
I tested this by enabling flow control and running this logger in a tight loop:

```python
for _ in range(100000): a.error('test')
```

...and I still couldn't get the socket to block in my testing after letting this run.